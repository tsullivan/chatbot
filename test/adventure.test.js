const request = require('supertest');
const { getServer } = require('../src/server');
const { utilFactory } = require('./utils');

const app = getServer();
const agent = request.agent(app);
const { handshake } = utilFactory(agent);

describe('adventure', () => {
  test('start', async () => {
    await handshake();

    const { body } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'play adventure' });
    expect(body.message).toMatchSnapshot();
  });

  test('exit fail', async () => {
    const { body: exitFail } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'exit' });
    expect(exitFail.message).toMatchSnapshot();
  });

  test('cave', async () => {
    const { body: caveLocation } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'cave' });
    expect(caveLocation.message).toMatchSnapshot();
  });

  test('cave look', async () => {
    const { body: caveLook } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'look' });
    expect(caveLook.message).toMatchSnapshot();
  });

  test('take_handle', async () => {
    const { body: takeHandle } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'take_handle' });
    expect(takeHandle.message).toMatchSnapshot();
  });

  test('cave exit', async () => {
    const { body: startLocation } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'exit' });
    expect(startLocation.message).toMatchSnapshot();
  });

  test('castle', async () => {
    const { body: castleLocation } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'castle' });
    expect(castleLocation.message).toMatchSnapshot();
  });

  test('castle sleep fail', async () => {
    const { body: sleep1 } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'sleep' });
    expect(sleep1.message).toMatchSnapshot();
  });

  test('use_window_handle', async () => {
    const { body: useHandle1 } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'use_window_handle' });
    expect(useHandle1.message).toMatchSnapshot();
  });

  test('castle look 1', async () => {
    const { body: lookCastle } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'look' });
    expect(lookCastle.message).toMatchSnapshot();
  });

  test('use_window_handle', async () => {
    const { body: useHandle2 } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'use_window_handle' });
    expect(useHandle2.message).toMatchSnapshot();
  });

  test('castle look 2', async () => {
    const { body: lookCastle } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'look' });
    expect(lookCastle.message).toMatchSnapshot();
  });

  test('castle sleep success', async () => {
    const { body: sleep2 } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'sleep' });
    expect(sleep2.message).toMatchSnapshot();
  });

  test('score', async () => {
    const { body: score } = await agent
      .post('/chat')
      .send({ format: 'user', message: 'score' });
    expect(score.message).toMatchSnapshot();
  });
});
