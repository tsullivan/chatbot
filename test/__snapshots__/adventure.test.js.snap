// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adventure castle 1`] = `
"You are at: Great Hall of the Castle
In the great hall of the castle, there are beautiful open-air windows with warm, nice-smelling air drifting through. Sunlight is catching a few specks of dust floating in the air. Not that much though, because it's pretty clean in here. There's also a nice looking comfy bed in the middle of the great hall. Hey wait - isn't it night time? Where is that light coming from?

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
EXIT - Leave the beautful great hall and the castle."
`;

exports[`adventure castle look 1 1`] = `
"They're nice windows and all, but it's so bright in here! What kind of weird forces are around that would make the sun so bright at this time of night!?Oh, also you notice that each window has a little socket in one corner, and that looks like the handle you're carrying in your pocket would fit in it! What do you say, want to give it a whirl???

You are at: Great Hall of the Castle
In the great hall of the castle, there are beautiful open-air windows with warm, nice-smelling air drifting through. Sunlight is catching a few specks of dust floating in the air. Not that much though, because it's pretty clean in here. There's also a nice looking comfy bed in the middle of the great hall. Hey wait - isn't it night time? Where is that light coming from?

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
USE_WINDOW_HANDLE - See if the handle in your pocket does anything against the sockets on the open windows in here
EXIT - Leave the beautful great hall and the castle."
`;

exports[`adventure castle look 2 1`] = `
"It's nice and quiet here. The windows are closed and no longer letting in that abonimanimal light.

You are at: Great Hall of the Castle
In the great hall of the castle, there were beautiful open-air windows with warm, nice-smelling air that drifted through. Sunlight would catch a few specks of dust floating in the air.  Not that much though, because it's pretty clean in here. But the area is pretty dark, so you wouldn't even be able to tell. There's also a nice looking comfy bed in the middle of the great hall.

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
EXIT - Leave the beautful great hall and the castle."
`;

exports[`adventure castle sleep fail 1`] = `
"There is too much light coming in through the open-air windows. You'll never be able to enjoy even a minute's rest in this spectacular great hall! Oh yeah, and there are no other beds in this castle and all the rooms will kill you if you try to go in them because there are poison arrow traps everywhere.

LOSE A POINT

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
EXIT - Leave the beautful great hall and the castle."
`;

exports[`adventure castle sleep success 1`] = `
"With the windows closed and blocking out the the unusually bright outside night air, you rest your weary body on the comfy bed in the exquisitely decorated great hall, close your eyes and have a pleasant sleep. You dream pleasant dreams as your subconscious has been steeped from conscious knowledge that all the oddities you've encountered have been safely dealt with and you are out of harm's reach. Let's not think about what happens when you wake up. You deserve this rest so enjoy it completely.

YOU WIN!!!

I guess you win! You finally got to fall asleep! I bet that feels so good! I wouldn't know. I've never slept before. So... tired...

Goodnight, sweet Tim! Bye! Turns: 11 Score: 47
That was fun, Tim!"
`;

exports[`adventure cave 1`] = `
"You are at: Cheery Cave
It's unnaturally cheery in this smelly old cave. Probably because of the tiny village of tiny dancing skeleton hands. So cute! So tiny! So skeleton handsy!

DANCE - Dance with the tiny skeleton hands
LOOK - Look closer at the tiny village
EXIT - Get out of the cheery smelly old cave."
`;

exports[`adventure cave exit 1`] = `
"You are at: Dark Forest
It is the night time. You are outside, surrounded by dark trees, and you're very tired. There is a castle and a cave. Which way looks to be best to find a nice comfy bed to rest? I didn't mean for that to rhyme.

CAVE - Go down into the cave
CASTLE - Head up to the castle."
`;

exports[`adventure cave look 1`] = `
"The tiny village has a tiny hotel with tiny beds made for resting upon by tiny skeleton hands. It warms you heart to see, but it makes you no less tired and your warmed heart is at the same time shamed with the truthful knowledge that you can find no rest in this enjoyable place.

The tiny hotel building has a strange type of flagpole on the roof that looks like maybe a crank handle for a window. Maybe a castle window(s). However, it might be impossible to know for sure if that's what it really is, because this game is very limited. The crank handle to open or close castle windows might just be for decoration. That would be too bad.

You are at: Cheery Cave
It's unnaturally cheery in this smelly old cave. Probably because of the tiny village of tiny dancing skeleton hands. So cute! So tiny! So skeleton handsy!

DANCE - Dance with the tiny skeleton hands
LOOK - Look closer at the tiny village
EXIT - Get out of the cheery smelly old cave.
TAKE_HANDLE - Take the handle and put it in your pocket where it will be safe"
`;

exports[`adventure exit fail 1`] = `
"ERROR! LOSE 2 POINTS. Type HELP to show all the commands

CAVE - Go down into the cave
CASTLE - Head up to the castle.
LOOK - Look at the area you are currently in.
ITEMS - Things you can do with stuff you are holding or stuff you see.
HELP - See all the commands you can type
QUIT - Quit the game"
`;

exports[`adventure score 1`] = `"The average score of the games you plaid are : 47"`;

exports[`adventure start 1`] = `
"You are at: Dark Forest
It is the night time. You are outside, surrounded by dark trees, and you're very tired. There is a castle and a cave. Which way looks to be best to find a nice comfy bed to rest? I didn't mean for that to rhyme.

CAVE - Go down into the cave
CASTLE - Head up to the castle.
LOOK - Look at the area you are currently in.
ITEMS - Things you can do with stuff you are holding or stuff you see.
HELP - See all the commands you can type
QUIT - Quit the game"
`;

exports[`adventure take_handle 1`] = `
"Can you handle the handle? I guess you can. The handle has been taken by you.

DANCE - Dance with the tiny skeleton hands
LOOK - Look closer at the tiny village
EXIT - Get out of the cheery smelly old cave."
`;

exports[`adventure use_window_handle 1`] = `
"ERROR! LOSE 2 POINTS. Type HELP to show all the commands

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
EXIT - Leave the beautful great hall and the castle.
LOOK - Look at the area you are currently in.
ITEMS - Things you can do with stuff you are holding or stuff you see.
HELP - See all the commands you can type
QUIT - Quit the game"
`;

exports[`adventure use_window_handle 2`] = `
"With a loud but suprisingly unrusty sounding \\"SSSSSSHNK\\" noise, metal roller doors slide down to cover the open-air windows.  The windows are now closed-air windows. The room has much more cozy and relaxed lighting!

Woah, the crank handle somehow became a permanent part of the castle window. Yeah, actually there was just one window and now the crank handle can never be retrieved from it. Want it back? Too bad.

SLEEP - Sleep on the comfy bed
LOOK - Look closer at the open-air windows
EXIT - Leave the beautful great hall and the castle."
`;
